Feature 23 scatter (#505)
dtcenter/METplotpy: docs/Users_Guide/scatter.rst

************
Scatter Plot
************

Description
===========
The scatter plot is useful for illustrating relationships between pairs of continuous variables.

> This plot was developed to support plotting MPR (matched pair) data from the MET point-stat tool.

Reviewer: ```suggestion
This plot was developed to support plotting MPR (matched pair) data from the MET Point-Stat tool.
```
Suggesting capitalization to match MET documentation

> **NOTE:** This MET output data must first be reformatted into a format that can be read in by the scatter plot code.

Reviewer: Suggest using the note directive (.. note:) to be consistent with other notes in the documentation.

Author: done

This reformatting was accomplished through the METdataio METreformat module. The reformatted data
consists solely of MPR linetype data and all the column headers are labelled according to the

> MPR linetype column names specified in the MET User's Guide, Table 11.20 (Point-stat tool).

Reviewer: ```suggestion
MPR linetype column names specified in the Point-Stat section of the MET User's Guide in `Table 11.20 <https://metplus.readthedocs.io/projects/met/en/latest/Users_Guide/point-stat.html#table-ps-format-info-seeps>`_ "Format information for SEEPS (Stable Equitable Error in Probability Space) output line type".
```
Suggest adding a link and more information, like the table name, in case the link should break, which hopefully it wouldn't, but just in case.

In addition to selecting the two continuous variables to plot, the points in the scatter plot are colored based on
their value.  These colors are determined by specifying a colormap in the user configuration file.

Example
=======

Sample Data
-----------

> The data is text output from the MET point-stat tool. The data is arranged in columnar format and reformatted by the

Reviewer: ```suggestion
The data is text output from the MET Point-Stat tool. The data is arranged in columnar format and reformatted by the
```

METdataio METreformat module.

The sample data for creating the example scatter plot is available in the
`METplotpy <https://github.com/dtcenter/METplotpy>`_ repository, where the scatter plot tests are located:

>     *$METPLOTPY_BASE/metplotpy/test/scatter/reformatted_data_for_scatter.data*

Reviewer: ```suggestion
    *$METPLOTPY_BASE/test/scatter/reformatted_data_for_scatter.data*
```
It looks like test is at the same level as metplotpy and not a subdir under metplotpy.

*$METPLOTPY_BASE* is the directory where the METplotpy code is saved:

e.g.

*/usr/path/to/METplotpy* if the source code was cloned or forked from the Github repository

or

*/usr/path/to/METplotpy-x.y.z* if the source code was downloaded as a zip or gzip'd tar file from the Release link of
the Github repository.  The *x.y.z* is the release number.

> **NOTE**:

Reviewer: Suggest using the note directive (.. note:) to be consistent with other notes in the documentation.

Author: done

> The MPR linetype data produced by the MET point-stat tool has been reformatted via the METdataio METreformat

Reviewer: ```suggestion
The MPR linetype data produced by the MET Point-Stat tool has been reformatted via the METdataio METreformat
```

module.

Configuration Files
-------------------

The scatter plot utilizes YAML configuration files to indicate where input
data is located and to set plot attributes. YAML is a recursive
acronym for "YAML Ain't Markup Language" and according to
`yaml.org <https://yaml.org>`_, it is a
"human-friendly data serialization language".
It is commonly used for configuration files and in applications where
data is being stored or
transmitted. Two configuration files are required. The first is a default
configuration file, **scatter_defaults.yaml**, which is found in the
*$METPLOTPY_BASE/metplotpy/plots/config* directory. All default
configuration files are located in the
*$METPLOTPY_BASE/metplotpy/plots/config* directory.
*$METPLOTPY_BASE* is base directory where the
METplotpy source code has been saved. **Default configuration files are
automatically loaded by the plotting code and do not need to be explicitly
specified when generating a plot**. In addition, the default configuration file
**DOES NOT require any modifications**.

The second required configuration file is a user-supplied “custom”

> configuration file. This file is used to customize/over ride the default

Reviewer: ```suggestion
configuration file. This file is used to customize/override the default
```

settings in the **scatter_defaults.yaml** file. The custom configuration

> file can contain only those settings that will over ride the default settings in the

Reviewer: ```suggestion
file can contain only those settings that will override the default settings in the
```
